deploy/kubernetes/elastic-agent-kustomize/ksm-hints/elastic-agent-standalone/kustomization.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - ../../default/elastic-agent-standalone
+
+patches:
+- patch: |-
+    apiVersion: apps/v1
+    kind: DaemonSet
+    metadata:
+      name: elastic-agent-standalone
+      namespace: kube-system
+      labels:
+        app: elastic-agent-standalone
+    spec:
+      template:
+        spec:
+          initContainers:
+            - name: k8s-templates-downloader
+              image: docker.elastic.co/beats/elastic-agent:9.0.0
+              command: ['bash']
+              args:
+                - -c
+                - >-
+                  mkdir -p /usr/share/elastic-agent/state/inputs.d &&
+                  curl -sL https://github.com/elastic/elastic-agent/archive/9.0.tar.gz | tar xz -C /usr/share/elastic-agent/state/inputs.d --strip=5 "elastic-agent-9.0/deploy/kubernetes/elastic-agent-standalone/templates.d"
```

Comment:
does this url https://github.com/elastic/elastic-agent/archive/9.0.tar.gz exist?

Reply:
No it does not exist, you are right. This is something that we always do,  we can not escape.

The caclulcation of the version is made based on the makefile https://github.com/elastic/elastic-agent/pull/5643/files/d1051cc2b11833d31df535b98d074ad76a3eea83#diff-18ea46196e4e61745238c819aae128261e8c81b1025bc9e455092d3e9f22817bR123

So we can merge this PR once the image is ready , or someone first to produce the manifest, change manually and then reapply

I dont have any preference

Reply:
I am wondering how difficult would be to include these templates in the agent image (or have a separate wolfi-based flavor that includes them in the first place?!). The whole `deploy/kubernetes/elastic-agent-standalone/templates.d` directory is 960KB in total which imo is negligible compared to the image size of agent. That said, it is not my intention to postpone this PR, so if this is how it is being done so far, I don't see any harm of doing it for one more time. But let's discuss how we could get away from this in the future

Reply:
We have a CI (see the command here https://github.com/elastic/elastic-agent/blob/e06e78666dc774b60ec0a2d398ec320a32feb601/.buildkite/scripts/steps/sync-k8s.sh#L26) which actually produces those templates under `deploy/kubernetes/elastic-agent-standalone/templates.d`.

Initially the idea was that we dont want to bundle the autodiscovery template process update with the actual beats release cycle. But considering the fact that we dont update them so often we can recosnider it.

It makes sense overal what you say

Reply:
https://github.com/elastic/elastic-agent/issues/5661 for tracking this

(FYI we have a META issue for hints [here](https://github.com/elastic/elastic-agent/issues/3063))